source/exe/main.cc

```diff
@@ -33,10 +35,36 @@ class ProdComponentFactory : public ComponentFactory {
 
 } // Server
 
+namespace {
+
+int validateConfig(OptionsImpl& options, Server::ProdComponentFactory& component_factory,
+                   const LocalInfo::LocalInfoImpl& local_info) {
+  Server::ValidationHotRestart restarter;
+  Logger::Registry::initialize(options.logLevel(), restarter.logLock());
+  Stats::HeapRawStatDataAllocator alloc;
+  Stats::ThreadLocalStoreImpl stats_store(alloc);
```

Comment:
Do we need TLS during validation? Or is it possible to complete all the initialization+validation on just the entry thread? If so, we could use a simpler `stats_store`, e.g. `IsolatedStoreImpl`.